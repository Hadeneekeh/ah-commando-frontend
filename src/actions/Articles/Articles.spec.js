import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import { getHomePageArticles, getArticle, getEditorsChoice } from './';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const url = 'https://a-haven-staging.herokuapp.com/api/v1';

describe('Home Page Articles Actions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });

  afterEach(() => {
    nock.cleanAll();
  });

  const response = {
    data: [
      {
        title: 'Example title 1',
        body: 'Some test',
      },
      {
        title: 'Example title 2',
        body: 'Some test',
      },
      {
        title: 'Example title 3',
        body: 'Some test',
      },
    ],
  };

  it('Should get the HomePage Articles', () => {
    nock(url)
      .get('/articles/categories/article')
      .reply(200, response);

    return store.dispatch(getHomePageArticles()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('Should get the featured Articles', () => {
    nock(url)
      .get('/articles/categories/article/featured')
      .reply(200, response);

    return store.dispatch(getArticle()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });


  it('Should get the Editors choice Articles', () => {
    nock(url)
      .get('/articles?limit=5')
      .reply(200, response);

    return store.dispatch(getEditorsChoice()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
