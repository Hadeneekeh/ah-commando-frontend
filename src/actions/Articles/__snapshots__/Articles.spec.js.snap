// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Articles Actions Should get the Editors choice Articles 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "body": "Some test",
          "title": "Example title 1",
        },
        Object {
          "body": "Some test",
          "title": "Example title 2",
        },
        Object {
          "body": "Some test",
          "title": "Example title 3",
        },
      ],
    },
    "type": "GET_EDITORS_CHOICE",
  },
]
`;

exports[`Home Page Articles Actions Should get the HomePage Articles 1`] = `Array []`;

exports[`Home Page Articles Actions Should get the featured Articles 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "body": "Some test",
          "title": "Example title 1",
        },
        Object {
          "body": "Some test",
          "title": "Example title 2",
        },
        Object {
          "body": "Some test",
          "title": "Example title 3",
        },
      ],
    },
    "type": "GET_FEATURED",
  },
]
`;
